src/Featurizers/Components/UnitTests/FilterFeaturizerImpl_UnitTest.cpp

```diff
@@ -0,0 +1,552 @@
+        FilterTransformer                   transformer;
+
+        transformer.execute(std::make_tuple('a', 10), callback);
+        transformer.execute(std::make_tuple('b', 20), callback);
+
+        CHECK(
+            results == Results{
+                typename Results::value_type('a', 10, 10),
+                typename Results::value_type('b', 20, 20)
+            }
+        );
+    }
+
+    SECTION("Delta") {
+        FilterTransformer                   transformer(100);
+
+        transformer.execute(std::make_tuple('a', 10), callback);
+        transformer.execute(std::make_tuple('b', 20), callback);
+
+        CHECK(
+            results == Results{
+                typename Results::value_type('a', 10, 110),
+                typename Results::value_type('b', 20, 120)
+            }
+        );
+    }
+}
+
+TEST_CASE("SimpleSumTransformer - Serialization") {
+    // ----------------------------------------------------------------------
+    using FilterTransformer =
+        Components::FilterTransformer<
+            SimpleSumTransformer,
+            std::tuple<char, int>,
+            1
+        >;
+
+    // ----------------------------------------------------------------------
+
+    FilterTransformer const                 transformer(10);
+
+    CHECK(transformer == FilterTransformer(10));
+    CHECK(transformer != FilterTransformer());
+
+    NS::Archive                             out;
+
+    transformer.save(out);
+
+    NS::Archive::ByteArray const            bytes(out.commit());
+    NS::Archive                             in(bytes);
+
+    FilterTransformer const                 other(in);
+
+    CHECK(other == transformer);
+}
+
+class TupleOutTransformer : public NS::StandardTransformer<int, std::tuple<int, int>> {
+public:
+    // ----------------------------------------------------------------------
+    // |  Public Types
+    using BaseType                          = NS::StandardTransformer<int, std::tuple<int, int>>;
+
+    // ----------------------------------------------------------------------
+    // |  Public Methods
+    TupleOutTransformer(void) = default;
+    ~TupleOutTransformer(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(TupleOutTransformer);
+
+    void save(NS::Archive &) const override {}
+
+private:
+    // ----------------------------------------------------------------------
+    // |  Private Methods
+    void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override {
+        callback(std::make_tuple(input, input));
+    }
+};
+
+TEST_CASE("TupleOutTransformer") {
+    // ----------------------------------------------------------------------
+    using FilterTransformer =
+        Components::FilterTransformer<
+            TupleOutTransformer,
+            std::tuple<int, char>,
+            0
+        >;
+
+    using Results                           = std::vector<std::tuple<int, char, int, int>>;

```

Comment:
If I understood the original test transformer, it takes in a value and outputs a tuple. It looks like the filter featurizer is then unpacking that tuple for a final result `std::tuple<int, char, int, int>`. Is there ever a situation where we woudln't want that and we would want the result to be `std::tuple<int, char, std::tuple<int, int>>`? I can't think of any off of the top of my head, but it is something I thought about.

Reply:
We could go this route, but then we would run into downstream composability problems. For example, it would be difficult to Filter with the second element of `tuple<int, int>` via the third element of `tuple<int, char, tuple<int, int>>`.